cmd/argocd/commands/app.go

```diff
@@ -384,6 +384,8 @@ func setAppOptions(flags *pflag.FlagSet, app *argoappv1.Application, appOpts *ap
 			app.Spec.Source.TargetRevision = appOpts.revision
 		case "values":
 			setHelmOpt(&app.Spec.Source, appOpts.valuesFiles)
+		case "directory-recurse":
+			app.Spec.Source.Directory = &argoappv1.ApplicationSourceDirectory{Recurse: true}
```

Comment:
This provides the ability to set directory to recurse, it looks like we don't provide the ability to unset it.

Also, we also prefer to nil out the the directory field entirely when the directory field is nothing but the already default value.

Basically the desired behavior is if the payload is this:
```yaml
spec:
  source:
    repoURL: https://github.com/argoproj/argocd-example-apps.git
    targetRevision: HEAD
    path: guestbook
    directory:
      recurse: false
```

Then it should get simplified to this:
```yaml
spec:
  source:
    repoURL: https://github.com/argoproj/argocd-example-apps.git
    targetRevision: HEAD
    path: guestbook
```

Because source.directory.false is already the default behavior. The benefit of doing this, is that it lets the user switch between application source types easily. For example they can switch between kustomize and plain yaml directory entirely via git.

Reply:
Let me check what happens. I thought that it got nulled out when you omitted `--directory-recurse`

Reply:
Behaves as you requested.

Upsert your app:

```
$ argocd app create guestbook --path guestbook --repo https://github.com/alexec/argocd-example-apps.git --dest-server https://localhost:6443  --dest-namespace default  --insecure --plaintext --upsert --directory-recurse
application 'guestbook' created
$ argocd app sync guestbook --insecure --plaintext
...
$ curl http://localhost:4000/api/v1/applications/guestbook | jq '.spec.source'
...
{
  "repoURL": "https://github.com/alexec/argocd-example-apps.git",
  "path": "guestbook",
  "directory": {
    "recurse": true
  }
}
```

Remove flag:

```
$ argocd app create guestbook --path guestbook --repo https://github.com/alexec/argocd-example-apps.git --dest-server https://localhost:6443  --dest-namespace default  --insecure --plaintext --upsert 
application 'guestbook' created               
$ argocd app sync guestbook --insecure --plaintext --prune
...
$ curl http://localhost:4000/api/v1/applications/guestbook | jq '.spec.source'
...
{
  "repoURL": "https://github.com/alexec/argocd-example-apps.git",
  "path": "guestbook"
}
```

Reply:
Not directory node is removed.